autogen/std_modelparameters.m

```diff
@@ -0,0 +1,106 @@
+% std_modelparameters.m
+%
+% This file sets the mp struct to have model parameters defined on it.
+% First it parses model.json, then reads modelparameters.m, and finally
+% computes additional fields from the previous settings.
+%
+% Copyright 2019 Audio Logic
+%
+% THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+% INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+% PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+% FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+% ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+%
+% Dylan Wickham
+% Audio Logic
+% 985 Technology Blvd
+% Bozeman, MT 59718
+% openspeech@flatearthinc.com
+%% Read the config 
+config_path = [modelPath filesep 'model.json'];
+config = jsondecode(fileread(config_path));
+
+mp.modelName = config.devices(1).name;
+
+%% Set Audio Data
+if isfield(config.system, "sampleClockFrequency")
+    mp.Fs = config.system.sampleClockFrequency; 
+else
+    mp.Fs = 48000;
+end
+
+if isfield(config.system, "numberOfChannels")
+    mp.nChannels = config.system.numberOfChannels;
+else
+    mp.nChannels = 2;
+end
+% Set the data type for audio signal
+mp.signed = config.system.inputDataType.signed;
+mp.W_bits = config.system.inputDataType.wordLength;  % Word length
+mp.F_bits = config.system.inputDataType.fractionLength;  % Number of fractional bits in word
+
+%% Configure target system
+if isfield(config.system, "target")
+    mp.target_system = config.system.target; 
+else
+    mp.target_system = "audiomini";
+end
+
+%% Parse registers
+mp.register = parseregisters(config);
+
+modelparameters
+
+%% Computed parameters
+% Set sample period for audio source
+mp.Ts = 1/mp.Fs; 
+
+%Set the FPGA system clock frequency (frequency of the FPGA fabric)
+% The system clock frequency should be an integer multiple of the Audio codec AD1939 Mclk frequency (12.288 MHz)
+
+if isfield(mp, "fastsim_flag") == 0 || mp.fastsim_flag == 0
+    if isfield(config.system, "systemClockFrequency")
+        mp.Fs_system = config.system.systemClockFrequency;        % System clock frequency in Hz of Avalon Interface  Mclk*8 = 12.288MHz*8=98304000
+    else
+        mp.Fs_system = 98304000;
+    end
+else
```

Comment:
Is this needed anymore now that we generate a default `model.json` with all defaults populated?

Reply:
That particular line isn't. Some of that stuff as it relates to fastsim, etc can as I think defaults for Autogen stuff in `modelparameters.m` are fine to support by making the optional.

Reply:
This has been updated. Only `modelparameters.m` settings have a default